SandboxTest/Main.cpp

```diff
@@ -202,30 +204,59 @@ int wmain (int argc, wchar_t *argv[]) {
 
     // create loader in a separate "low integrity" dllhost.exe process
     CComPtr<IImage3dFileLoader> loader;
-    {
-        std::wcout << L"Creating loader " << progid.m_str << L" in low-integrity mode...\n";
-        LowIntegrity low_integrity;
-        PerfTimer timer("CoCreateInstance", profile);
-        CHECK(loader.CoCreateInstance(clsid, nullptr, CLSCTX_LOCAL_SERVER | CLSCTX_ENABLE_CLOAKING));
-    }
+    CComPtr<IStream>            stream;
+    std::atomic<bool>           loaded = { false };
+
+    std::thread loader_thread([&]() {
```

Comment:
A 40-line lambda seem overly long too me. Would it be possible to e.g. instead move the code into a separate function, so that the lambda can be kept small? Also, we should probably make threaded communication optional through a command-line argument of similar.

Reply:
Agree.

Reply:
Done.